packages/haul-core/src/utils/getReactNativeVersion.ts

```diff
@@ -0,0 +1,14 @@
+import path from 'path';
+
+export default function getReactNativeVersion(cwd: string): string | null {
+  try {
+    const { version } = require(path.join(
+      cwd,
+      'node_modules/react-native/package.json'
+    ));
```

Comment:
you can use 
```js
require(require.resolve('react-native/package.json', {paths: [cwd]}))
``` 
for future compatibility with Yarn PnP (still far from happening though)